src/@next/Upload/Upload.stories.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { useState } from 'react';
+import { Story, Meta } from '@storybook/react';
+import { BaseContainer } from '../../Layout/GlintsContainer/GlintsContainer';
+import { Upload, UploadProps } from './Upload';
+
+export default {
+  title: '@next/Upload',
+  component: Upload,
+  decorators: [Story => <BaseContainer>{Story()}</BaseContainer>],
+  argTypes: {},
+} as Meta;
+
+const Template: Story<UploadProps> = args => {
+  const [file, setFile] = useState<File | null>(null);
+
+  return (
+    <>
+      <Upload {...args} file={file} setFile={setFile} />
+    </>
+  );
+};
+export const Interactive = Template.bind({});
+Interactive.args = {};
+
+const LoadingTemplate: Story<UploadProps> = args => {
+  const [file, setFile] = useState<File | null>(null);
+
+  return (
+    <>
+      <Upload {...args} file={file} setFile={setFile} loading={true} />
+    </>
+  );
+};
```

Comment:
actually, I shouldn't even need to make `loading` as a prop here... I just make this so that the designers can see the loading component for QC...

imo though, the loading component is not that relevant (almost always never be seen or you just see a 0.01 flash of it) as the file should be less than 5mb, and the time should almost always be negligible (<0.1sec)...
the time for uploading (when submitting the form) might vary depending on internet connection etc, but this loading is NOT for uploading (submit the form containing file) to the server, but just simply 'set the file path';  larger files like video might be longer due to metadata etc

Reply:
will remove exposing loading as prop as qc is completed

Reply:
nvm im not removing it as it's still used for testing... when testing, I can't see loading state hmm.... it automatically goes to image state, even when I'm not using await or adding any time delays